File: silver/1920/python/main.py

if (__name__ == "__main__"):
    num = int(input())
    key = list(map(int, input().split()))[:num]
    base = dict()

    for data in key:
        base[data] = 1

    num = int(input())
    value = list(map(int, input().split()))[:num]

    for data in value:
        if (base.get(data) == 1):
            print(1)
        else :
            print(0)

File: silver/1920/c++/main.cpp
#include <iostream>
#include <map>

int main()
{
    std::ios::sync_with_stdio(false);
    std::cin.tie(NULL);
    
    int num, data;
    std::cin >> num;
    std::map<int, int> base;
    for (int i = 0; i < num; i++) {
        std::cin >> data;
        base[data] = 1;
    }

    std::cin >> num;

    for (int i = 0; i < num; i++) {
        std::cin >> data;
        if (base.count(data)) {
            std::cout << "1\n";
        } else {
            std::cout << "0\n";
        }
    }

    return 0;
}